### PR TITLE
Fix/upsert secrets issue 210

### PR DESCRIPTION
**Pull request by marboledacci**
The secrets update was not managed correctly as mentioned in #210. I'm fixing that here, and doing some updates to the tests.

### CODE REVIEW

File: tests/terraform_setup/ec2/cloudformation-templates/public-service.yml [hunk 1]
```diff
@@ -69,11 +69,16 @@ Resources:
...
-        Fn::If:
-          - 'HasCustomRole'
-          - !Ref 'Role'
-          - !Ref "AWS::NoValue"
+        Fn::ImportValue:
+            !Join [':', [!Ref 'StackName', 'EC2Role']]
+      #   Fn::If:
+      #     - 'HasCustomRole'
+      #     - !Ref 'Role'
+      #     - !Ref "AWS::NoValue"
```
Comment (david-montano-circleci) [inline tests/terraform_setup/ec2/cloudformation-templates/public-service.yml:81]: Why is this if commented out?

File: tests/terraform_setup/ec2/terraform.tf [hunk 2]
```diff
@@ -64,6 +61,7 @@ resource "aws_cloudformation_stack" "ecs_service" {
   depends_on = [
     aws_cloudformation_stack.vpc,
     aws_ecr_repository.demo-app-repository,
+    # aws_iam_role.parameter_store_read_role
```
Comment (david-montano-circleci) [inline tests/terraform_setup/ec2/terraform.tf:64]: Why was this removed?

File: tests/terraform_setup/ec2/terraform.tf [hunk 3]
```diff
@@ -28,7 +25,7 @@ locals {
   aws_ecs_service_stack_name = "${var.aws_resource_prefix}-svc-stack"
 
   # The name of the ECR repository to be created
-  aws_ecr_repository_name = var.aws_resource_prefix
+  aws_ecr_repository_name = "${var.aws_resource_prefix}"
```
Comment (david-montano-circleci) [inline tests/terraform_setup/ec2/terraform.tf:28]: In theory there is no need to put this variable inside a string.

File: tests/terraform_setup/ec2/variables.tf [hunk 4]
```diff
@@ -1,7 +1,7 @@
-variable "aws_access_key" {}
-variable "aws_secret_key" {}
+# variable "aws_access_key" {}
+# variable "aws_secret_key" {}
```
Comment (david-montano-circleci) [inline tests/terraform_setup/ec2/variables.tf:2]: These can be removed if I understand correctly. I am supposing this uses the role assigned to the EC2 instance where the pipeline runs.

File: tests/terraform_setup/fargate/variables.tf [hunk 5]
```diff
@@ -1,7 +1,7 @@
-variable "aws_access_key" {}
-variable "aws_secret_key" {}
+# variable "aws_access_key" {}
+# variable "aws_secret_key" {}
```
Comment (david-montano-circleci) [inline tests/terraform_setup/fargate/variables.tf:2]: These can be removed alongside the session token.

File: tests/terraform_setup/fargate/terraform.tf [hunk 6]
```diff
@@ -28,7 +25,7 @@ locals {
   aws_ecs_service_stack_name = "${var.aws_resource_prefix}-svc-stack"
 
   # The name of the ECR repository to be created
-  aws_ecr_repository_name = var.aws_resource_prefix
+  aws_ecr_repository_name = "${var.aws_resource_prefix}"
```
Comment (david-montano-circleci) [inline tests/terraform_setup/fargate/terraform.tf:28]: Just like before, there is no need to use a string interpolation here, is there?

**Review by david-montano-circleci — CHANGES_REQUESTED** [review]

File: tests/terraform_setup/fargate/terraform.tf (hunk 6 above)
Comment (marboledacci) [reply tests/terraform_setup/fargate/terraform.tf:28]: No need, I had updated the name of the resources and let the string interpolation syntax in case it was required again.

File: tests/terraform_setup/ec2/variables.tf (hunk 4 above)
Comment (marboledacci) [reply tests/terraform_setup/ec2/variables.tf]: It uses OIDC authentication during the workflow, there's no need to pass the keys directly.

File: tests/terraform_setup/ec2/terraform.tf (hunk 2 above)
Comment (marboledacci) [reply tests/terraform_setup/ec2/terraform.tf]: The role was moved to the Cloudformation template

File: tests/terraform_setup/ec2/terraform.tf (hunk 3 above)
Comment (marboledacci) [reply tests/terraform_setup/ec2/terraform.tf:28]: As the other, there's no need, it's there for easier name changing if required.

**Review by david-montano-circleci — APPROVED** [review]
